fe/fe-core/src/main/java/org/apache/doris/planner/BrokerScanNode.java

```diff
@@ -395,10 +405,21 @@ protected void getFileStatus() throws UserException {
     }
 
     private void assignBackends() throws UserException {
+        Set<Tag> tags = Sets.newHashSet();
+        if (userIdentity != null) {
+            tags = Catalog.getCurrentCatalog().getAuth().getResourceTags(userIdentity.getQualifiedUser());
+            if (tags == UserProperty.INVALID_RESOURCE_TAGS) {
+                throw new UserException("No valid resource tag for user: " + userIdentity.getQualifiedUser());
+            }
+        } else {
+            LOG.debug("user info in BrokerScanNode should not be null, add log to observer");
```

Comment:
I think user info could be allowed null here in a BrokerScanNode used by query.

Reply:
Why? I think the user info must be set for both query and load. If not, may be we missed somewhere to set it.

Reply:
I see. I just want to see where do I miss to set the user info, so I add this debug log here.